lib/web_ui/lib/src/engine/canvaskit/text.dart

```diff
@@ -660,8 +633,50 @@ class CkParagraphBuilder implements ui.ParagraphBuilder {
     return properties;
   }
 
+  /// Determines if the given [text] contains any code points which are not
+  /// supported by the current set of fonts.
+  void _verifyFontsSupportText(String text) {
```

Comment:
This looks like an expensive method that's called on every `addText`. Would it be worth caching something here? For example, apps tend to repeatedly render the same pieces of text with the same style. In addition to (or instead of) caching, we could identify common cases that might need less processing.

Reply:
We have a benchmark for text layout that could help measure the impact.

Reply:
I am checking if all the text is ASCII first.